### ISSUE TITLE
Mono/C# crashes when thread model is set to Multi-Threaded

### ISSUE THREAD

**Issue by LaBlazer** [godotengine/godot#20673]
**Godot version:**
3.0.6 mono

**OS/device including version:**
Windows 10 17134, RX480

**Issue description:**
Project crashes without any stacktrace in the debugger, but works in single-threaded model. Also for some reason "0" is being print in the console when it crashes.

**Steps to reproduce/minimal reproduction project:**
1. Clone https://github.com/LaBlazer/stylized-planet-generator/tree/csharp
2. Change Thread Model to Multi-Threaded
3. Run project

**Comment by malbach** [comment]
Hi,
Can confirm the bug using [6c569c9](https://github.com/godotengine/godot/commit/6c569c90b666c7fb773eca3948fc76ba7a160a27) compiled with MONO 5.12.0
The 0s in the console come from errors while compiling the shader : [shader_compiler_gles3.cpp](https://github.com/godotengine/godot/blob/6c569c90b666c7fb773eca3948fc76ba7a160a27/drivers/gles3/shader_compiler_gles3.cpp#L731)
I did have more infos in the console (repeated 58 times) : 
```
0
SCRIPT ERROR: (null): Expected 'shader_type' at the beginning of shader.
          At: :1
ERROR: RasterizerStorageGLES3::_update_shader: Condition ' err != OK ' is true.
   At: drivers\gles3\rasterizer_storage_gles3.cpp:1760
```

It's to be noted that the bunch of 0s in the console are also present in the non-MONO version, but nothing else.

**Comment by LaBlazer** [comment]
The zeroes are probably caused by 0e29f7974b59e4440cf02e1388fb9d8ab2b5c5fd, it added a bunch of 0s all over the place, HDRs are unable to import because of it too... (fixed by https://github.com/godotengine/godot/pull/20636)

**Comment by neikeq** [comment]
This is a known problem. It's related to #16024. I'm figuring out a way to attach thread automatically without causing issues later.

**Comment by hpvb** [comment]
@LaBlazer can you elaborate on 0e29f79 adding 0s? I don't see any print_line or printfs or anything being added in that PR

**Comment by vnen** [comment]
@hpvb the problem with 0e29f79 is that some strings were being created with vector methods (in particular `push_back`) and added a null-terminator. Since you replaced those `push_back()` with the `+=` operator, now it actually adds the literal character `0` at the end instead of the null-terminator.

**Comment by LaBlazer** [comment]
@hpvb search for `+= '0'` in the commit...

**Comment by akien-mga** [comment]
See also #21725.

**Comment by akien-mga** [comment]
Might be fixed by #25927, please try again when we publish 3.1 beta 5 or later.

**Comment by hpvb** [comment]
This error is definitely a symptom fixed by that PR. I'm closing this. Feel free to reopen if it reappears.
